corehq/apps/case_importer/static/case_importer/js/excel_fields.js

```diff
@@ -0,0 +1,89 @@
+
+        self.caseFieldsInMenu = _(self.caseFieldSpecsInMenu).pluck('field');
+        self.caseFieldSuggestions = _.chain(self.caseFieldSpecs).where({discoverable: true}).pluck('field').value();
+        self.mappingRows = ko.observableArray();
+        self.removeRow = function (row) {
+            self.mappingRows.remove(row);
+        };
+        self.sanitizeCaseField = function (original_value) {
+            var value = original_value;
+            // space to underscore
+            value = value.replace(/\s/g, "_");
+            // remove other symbols
+            value = value.replace(/[^a-zA-Z0-9_\-]/g, "");
+            // remove xml from beginning of string. todo: why (36cafb01)?
+            value = value.replace(/^xml/i, "");
+            return value;
+        };
+        self.addRow = function (excelField) {
+            var row = {
+                excelField: ko.observable(excelField),
+                selectedCaseField: ko.observable(null),
+                customCaseField: ko.observable(excelField),
+                isCustom: ko.observable(false),
+            };
+            row.caseField = ko.computed(function () {
+                if (row.isCustom()) {
+                    return row.customCaseField();
+                } else {
+                    return row.selectedCaseField();
+                }
+            });
+
+            row.caseFieldSpec = ko.computed(function () {
+                return _(caseFieldSpecs).findWhere({field: row.caseField()}) || {};
+            });
+            row.hasDiscoverableSpecialField = ko.computed(function () {
+                return row.caseFieldSpec().description && row.caseFieldSpec().discoverable;
+            });
+            row.hasNonDiscoverableField = ko.computed(function () {
+                return row.caseFieldSpec().description && !row.caseFieldSpec().discoverable;
+            });
+            row.reset = function () {
+                var field = row.excelField();
+                field = field && self.sanitizeCaseField(field);
+                row.customCaseField(field);
+                if (!field || _(self.caseFieldsInMenu).contains(field)) {
+                    row.isCustom(false);
+                    row.selectedCaseField(field);
+                } else {
+                    row.isCustom(true);
+                    row.selectedCaseField(null);
+                }
+            };
+            row.reset();
+            row.caseFieldSuggestions = ko.computed(function () {
+                var field = row.caseField();
+                if (!field || _(self.caseFieldSuggestions).contains(field)) {
+                    return [];
+                }
+                var suggestions = _(self.caseFieldSuggestions).map(function (suggestion) {
+                    return {distance: Levenshtein.get(field, suggestion), field: suggestion};
+                }).filter(function (suggestion) {
+                    return suggestion.distance < 4;
```

Comment:
if the distance is 4 maybe we should do non case sensitive levenshtein? otherwise `OWNER_names` will be distance 6 away from `owner_name`. not sure how to best handle since i think the backend depends on the case sensitivity of the field. pretty minor point